doc/shapes/2025.12/customize_elasticsearch_index_to_work_with_sysml.adoc

```diff
@@ -0,0 +1,68 @@
+= Customize Elasticsearch index to work with SysML metamodel
+
+== Problem
+
+Sirius Web now provides a command in the omnibox of the `/projects` page to search elements in all the projects.
+This command is based on Elasticsearch, which contains indexed documents built from the semantic data of the projects.
+Sirius Web provides a default mechanism to create these documents, but it doesn't fit well the SysML metamodel:
+
+- Indices contain irrelevant information for SysON users (e.g. `Membership` instances).
+- Multi-level navigation is not possible (e.g. usage.definition.name), which works particularly poorly with SysML where all the links are reified.
+
+We need to provide a custom index document creation process, in order to support relevant queries for SysML users.
+The goal of this shape is to define these queries, the technical details of the serialization process will be discussed in a future ADR.
+
+== Prerequisites
+
+Sirius Web's search command relies on Elasticsearch https://www.elastic.co/docs/reference/query-languages/query-dsl/query-dsl-query-string-query[query string DSL].
+This language allows to search for documents (i.e. objects stored in the index) matching a given input.
+It supports wildcards, regular expressions, fuzziness, and provides constructs to search in specific fields, check ranges, or exclude some values.
+
+For example, the following query searches for document with a `name` field containing values starting with `MyName`, and the `sibling.age` field between 5 and 10:
+
+----
+name:MyName* AND sibling.age:[5 TO 10]
+----
+
+== Solution
+
+All the `Element` instances are stored in the index.
+
+The `name`, `shortName`, and `qualifiedName` attributes can be used to find elements.
+
+The following references can be navigated as part of the query:
+- `Element#owner`
+- `Element#ownedElement`
+- `Type#ownedSpecialization`
+- `Specialization#general`
+
+Navigation has a maximum depth of 1: for example, it is possible to access the owner of an element, but the owner of the owner of an element.
+This limitation is necessary to limit the number of fields stored in the index (by default the maximum number of fields is 1000), and to avoid false positive matches on unbounded queries (e.g. an indexed document can match a term if its n-th owner's name contains the term).
+
+The list below shows some example of queries that should be supported/not supported by SysON:
+
+----
+// Get the element that have a field that matches "mySearchValue"
+mySearchValue
+// Get the elements with a name starting with "my" in Package1
+name:my* owner.name:Package1
+// Get the Packages containing elements with a name starting with "my" or "part"
+@type:Package AND ownedElement.name:(my* OR part*)
+// Get the elements typed by a::MyType
```

Comment:
That means users will be forced to use the syntax `sysml::PartUsage`? They won't be able to use `PartUsage` directly?
We also use `Part` instead of `PartUsage` (same for all Usages) in the UI.
I don't know how it will be difficult/if it will be possible to allow ElasticSearch to let users use `Part` instead of `PartUsage`?

Reply:
I clarified the comment.
As you can see above, you can check a type by its name, without the `sysml::` prefix (e.g. `@type:Package`).
The query with `a::MyType` is checking the qualified name of the type, that's why it contains `::`.

I also added a paragraph about the attributes provided by Sirius Web (`@type`, `@label`, etc). To make it clear why there are `@` in some queries

Reply:
We can also remove the "Usage" from the type name. I clarified this part in the shape.